### DISCUSSION TITLE
Using `antigen` with `chezmoi`.

### DISCUSSION THREAD

**Discussion by Noahyt** [twpayne/chezmoi#2009] [Q&A] [answered]
I am new to the world of `chezmoi` and `zsh` customization in general. 

I know that `chezmoi` has the option to install external libraries using the `.chezmoiexternal.toml` file. 

How does this relate/conflict with using [antigen](https://github.com/zsh-users/antigen) to manage packages?

Is there a preferred way to integrate both tools? I've looked around a bit and the only reference I could find was [this](https://github.com/HHousen/dotfiles) dotfile.

**Comment by twpayne** [comment, marked as answer]
I'm not familiar with antigen (I use [vim-plug](https://github.com/junegunn/vim-plug)) but I would expect antigen and chezmoi externals to conflict if you attempt to use both simultaneously. If antigen updates files managed by chezmoi then chezmoi will detect that the file has been modified since chezmoi last wrote it and you will get a prompt the next time you use chezmoi to update the file. This is prevent chezmoi from blindingly overwriting any local changes that you might want to keep.

I would suggest the following approaches:
1. Either, use a chezmoi external to install antigen and then use antigen to manage your zsh plugins, as HHousen has done.
2. Or, use chezmoi externals for all zsh plugins, and don't use antigen at all.
Both approaches work, but should not be mixed.
